nntrainer/src/databuffer.cpp

```diff
@@ -106,6 +116,53 @@ bool DataBuffer::init(int mini_batch, unsigned int train_bufsize,
   return true;
 }
 
+int DataBuffer::init() {
+
+  int status = ML_ERROR_NONE;
+
+  if (!this->class_num) {
+    ml_loge("Error: number of class must be set");
+    for (int i = 0; i < NBUFTYPE; ++i)
+      validation[i] = false;
```

Comment:
Q: I wonder how this per-buffer validation value will be utilized. I think a single `initialized` variable may perform similar behaviors.

Reply:
There will be three cases (Training, Validation, Testing) with Label data. 
validation[0]: Training Set
validation[1]: Validation Set
validation[2]: Testing Set
validation[3]: Label data

Training and label is essential and others are optional. 
Depending on this data provided,  consequence procedure will be coducted.

Reply:
Ah, some buffers would be optional. Got it.